include/boost/geometry/strategies/closest_points/spherical.hpp

```diff
@@ -0,0 +1,112 @@
+// Boost.Geometry
+
+// Copyright (c) 2021, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_CLOSEST_POINTS_SPHERICAL_HPP
+#define BOOST_GEOMETRY_STRATEGIES_CLOSEST_POINTS_SPHERICAL_HPP
+
+
+//#include <boost/geometry/strategies/cartesian/azimuth.hpp>
+
+//#include <boost/geometry/strategies/cartesian/distance_projected_point.hpp>
+//#include <boost/geometry/strategies/cartesian/distance_pythagoras.hpp>
+//#include <boost/geometry/strategies/spherical/distance_pythagoras_box_box.hpp>
+//#include <boost/geometry/strategies/spherical/distance_pythagoras_point_box.hpp>
+//#include <boost/geometry/strategies/spherical/distance_segment_box.hpp>
+
+#include <boost/geometry/strategies/spherical/closest_points_pt_seg.hpp>
+
+#include <boost/geometry/strategies/detail.hpp>
+#include <boost/geometry/strategies/distance/detail.hpp>
+#include <boost/geometry/strategies/closest_points/services.hpp>
+
+#include <boost/geometry/strategies/distance/spherical.hpp>
+
+#include <boost/geometry/util/type_traits.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategies { namespace closest_points
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail
+{
+
+template <typename RadiusTypeOrSphere, typename CalculationType>
+class spherical
+    : public strategies::distance::detail::spherical<RadiusTypeOrSphere, CalculationType>
+{
+    using base_t = strategies::distance::detail::spherical<RadiusTypeOrSphere, CalculationType>;
+
+public:
+    spherical() = default;
+
+    template <typename RadiusOrSphere>
+    explicit spherical(RadiusOrSphere const& radius_or_sphere)
+        : base_t(radius_or_sphere)
+    {}
+
+    template <typename Geometry1, typename Geometry2>
+    auto closest_points(Geometry1 const&, Geometry2 const&,
+        distance::detail::enable_if_ps_t<Geometry1, Geometry2> * = nullptr) const
+    {
+        return strategy::closest_points::cross_track<CalculationType>(
+            base_t::radius());
+    }
+    
+};
+
+
+} // namespace detail
+#endif // DOXYGEN_NO_DETAIL
+
+template
+<
+    typename RadiusTypeOrSphere = double,
+    typename CalculationType = void
+>
+class spherical
```

Comment:
Here as well, why is base class in `detail` needed instead of defining membre functions here?

Reply:
The definition of base class in detail namespace is not needed.